src/main/java/com/example/controller/FileMenuController.java

```diff
@@ -18,12 +21,26 @@
 public class FileMenuController {
     @Autowired
     GraphController graphController;
+    @Autowired
+    AppController appController;
+    @FXML
+    MenuItem newFileButton;
     @FXML
     MenuItem importButton;
     @FXML
     MenuItem exportButton;
 
     private void bindButtons() {
+        ObjectProperty<ApplicationState> applicationState = appController.getApplicationStateProperty();
+        BooleanBinding isSimulation = applicationState.isEqualTo(ApplicationState.SIMULATING);
+
+        newFileButton.disableProperty().bind(isSimulation);
+
+        newFileButton.setOnAction(e -> {
+            graphController.getGraph().vertices().clear();
```

Comment:
I think that better option here would be to create new graph and set it, instead of clearing existing one. This approach may generate hidden erros, e.x. indexing vertices is not working as it should.

Reply:
You're right. Fixed.